components/mailer.rst

```diff
@@ -16,8 +16,127 @@ Installation
 
 .. include:: /components/require_autoload.rst.inc
 
+
+Introduction
+------------
+
+
 Usage
 -----
 
-We're currently working on the documentation of this component that was just
-added to Symfony. We'll publish it in a few days.
+The Mailer component has two main classes: a ``Transport`` and the ``Mailer`` itself::
+
+    use Symfony\Component\Mailer\Mailer;
+    use Symfony\Component\Mailer\Transport\Smtp\SmtpTransport;
+
+    $transport = new SmtpTransport('localhost');
+    $mailer = new Mailer($transport);
+    $mailer->send($email);
+
+Refer :doc:`Mime component </components/mime>` how to create `$email` object.
+
+Transport
+---------
+
+By default, the only transport available in the mailer component is Smtp.
+
+Below is the list of other popular providers with built in support.
+
+- Amazon SES  : symfony/amazon-mailer
+- Google Gmail : symfony/google-mailer
+- Mandrill : symfony/mailchimp-mailer
+- Mailgun : symfony/mailgun-mailer
+- Postmark : symfony/postmark-mailer
+- Sendgrid : symfony/sendgrid-mailer
+
+For example to use google's gmail as a transport you need to install symfony/google-mailer.
+
+.. code-block:: terminal
+
+    $ composer require symfony/google-mailer
+
+.. code-block:: php
+    use Symfony\Component\Mailer\Bridge\Google\Smtp\GmailTransport;
```

Comment:
Please add an empty line between `.. code-block:: php` and the actual code.

A very small reStructured Text tip (as it'll help you understand what we suggest). Code blocks start with `.. code-block:: LANGUAGE` followed by an empty line. Or if it's PHP and the previous sentence ended with a colon (like "See this example:"), you can omit the `.. code-block:: php` line and replace it with a double colon:

```diff
- See this example:
- 
- .. code-block:: php  
+ See this example::
  
      echo 'Hello'
```

Reply:
Thank you for clearing it.